app/src/main/java/com/infomaniak/swisstransfer/ui/components/SwissTransferTextField.kt

```diff
@@ -0,0 +1,177 @@
+/*
+ * Infomaniak SwissTransfer - Android
+ * Copyright (C) 2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.swisstransfer.ui.components
+
+import android.text.Html
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.*
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.infomaniak.swisstransfer.R
+import com.infomaniak.swisstransfer.ui.images.AppImages.AppIcons
+import com.infomaniak.swisstransfer.ui.images.icons.EyeCrossedOutSmall
+import com.infomaniak.swisstransfer.ui.images.icons.EyeSmall
+import com.infomaniak.swisstransfer.ui.theme.Margin
+import com.infomaniak.swisstransfer.ui.theme.SwissTransferTheme
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+@Composable
+fun SwissTransferTextField(
+    modifier: Modifier = Modifier,
+    isRequired: Boolean = true,
+    label: String? = null,
+    initialValue: String = "",
+    minLineNumber: Int = 1,
+    maxLineNumber: Int = if (minLineNumber > 1) Int.MAX_VALUE else 1,
+    keyboardType: KeyboardType = KeyboardType.Text,
+    errorMessageState: StateFlow<String?> = MutableStateFlow(null).asStateFlow(),
```

Comment:
This feels like passing an entier flow is not the correct way of doing things. We usually pass in callbacks to either read or update the state that's managed outside. Maybe the sole reason is lowering the coupling and keeping a high cohesion, I don't know.

If you want to exact explanation ask @sirambd and tell me if you go ask him because I'm interested as well

Reply:
Yes thanks I actually thought it was stange